src/py2spack/loading.py

```diff
@@ -20,166 +21,257 @@
     ".bz2",
 ]
 
-HTTP_STATUS_SUCCESS = 200
-HTTP_STATUS_NOT_FOUND = 404
-
-
-class APIError(Exception):
```

Comment:
Renamed APIError to PyProjectProviderQueryError. If we have multiple providers (pypi, github) it could make sense to have different errors for them...